lib/apiservers/engine/backends/container.go

```diff
@@ -1119,36 +1133,53 @@ func attachStreams(ctx context.Context, vc *viccontainer.VicContainer, clStdin i
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
+
 			err := copyStdErr(ctx, plClient, vc, clStderr)
 			if err != nil {
-				log.Errorf("container attach: stderr (%s): %s", vc.ContainerID, err.Error())
+				log.Errorf("stderr (%s) error: %s", vc.ContainerID, err.Error())
 			} else {
-				log.Infof("container attach: stderr (%s) done: %s", vc.ContainerID)
+				log.Infof("stderr (%s) done", vc.ContainerID)
 			}
 
+			// no need to take action if we are canceled
+			// as that means error happened somewhere else
+			if ctx.Err() == context.Canceled {
+				log.Infof("returning from stderr as context canceled somewhere else")
+				return
+			}
 			cancel()
+
 			errors <- err
 		}()
 	}
 
 	// Wait for all stream copy to exit
 	wg.Wait()
 
-	log.Infof("container attach:  cleaned up connections to %s.", vc.ContainerID)
+	// close the channel so that we don't leak (if there is an error)/or get blocked (if there are no errors)
+	close(errors)
+
+	log.Infof("cleaned up connections to %s. Checking errors", vc.ContainerID)
 	for err := range errors {
 		if err != nil {
+			// check if we got DetachError
+			if _, ok := err.(DetachError); ok {
```

Comment:
Will we get a `DetachError` from all of the `go func`s above or is it possible only 1 returns this error, while the other 2 return CDE?  Just asking out of curiosity.